### PR TITLE
🎟 Wrap event fixtures the way they're supposed to be

### PR DESCRIPTION
**Pull request by barlock**
**Related PRs**  
This PR is not dependent on any other PR

**What does this PR do?**  
I've finally figured out the fixture shenanigans! I split them up the way bolt does and added a helper to create real event fixtures. 

In the process, I simplified the api of the test receiver to match more closely the serverless tester which is generic and relies on the fixture generators for the "heavy lifting". Slightly more verbose, less wrapping to maintain. At some point we might be able to generate those helpers, but...  that's for later.

This also bumps two repos to 2.0.0, I'm like 99% sure this will Just Work™️ 

**Description of Changes**  
I think I did that above, I'll annotate the weird business. 

**What gif most accurately describes how I feel towards this PR?**

### CODE REVIEW

File: packages/bolt-interactions/package.json [hunk 1]
```diff
@@ -9,14 +9,19 @@
...
   "license": "Apache-2.0",
   "dependencies": {
-    "@slack/bolt": "^1.4.1",
     "@slack/types": "^1.3.0",
     "@slack/web-api": "^5.6.0",
     "@types/uuid": "^3.4.6",
     "uuid": "^3.3.3"
   },
+  "peerDependencies": {
+    "@slack/bolt": ">=1.0.0 <3.0.0"
```
Comment (barlock) [inline packages/bolt-interactions/package.json]: This caused weirdness actually trying to use it especially with mismatched types between 1 and 2.

File: packages/bolt-interactions/src/types/interaction.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { App, SlackAction, SlackActionMiddlewareArgs } from '@slack/bolt';
```
Comment (barlock) [inline packages/bolt-interactions/src/types/interaction.ts]: So it turns out that by default custom `.d.ts` files aren't built. 🤷‍♂ They're intended to be a "hint to the compiler". SO, to export types, they have to be `.ts` files. I made that change, and converted it to a module to remove the namespace.

File: packages/fixtures/package.json [hunk 3]
```diff
@@ -14,11 +14,14 @@
   },
   "license": "Apache-2.0",
   "dependencies": {
-    "@slack/bolt": "^1.5.0",
     "aws-lambda": "^1.0.5",
     "serverless-http": "^2.3.1"
   },
+  "peerDependencies": {
+    "@slack/bolt": ">=1.0.0 <3.0.0"
```
Comment (barlock) [inline packages/fixtures/package.json]: Same problem as above.

File: packages/bolt-interactions/src/types/interaction.ts (hunk 2 above)
Comment (barlock) [reply packages/bolt-interactions/src/types/interaction.ts]: This is not related to the pr, so, a bit of boy scouting 😅

Comment (tteltrab) [reply packages/bolt-interactions/src/types/interaction.ts]: It makes sense that `.t.s` files are excluded by default. If we want to include them shouldn't we just use a compiler flag? 

or I guess have a `main.d.ts` that we point the `types` property to in our `package.json` that grabs the rest.

https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package

File: packages/fixtures/src/fields.spec.ts [hunk 4]
```diff
@@ -15,15 +15,15 @@ describe('Defaults', () => {
...
+    expect(actions.blockButtonAction()).toEqual(
       expect.objectContaining({
         token: originalToken,
       }),
     );
 
     fields.token = overrideToken;
 
-    expect(events.blockButtonAction()).toEqual(
+    expect(actions.blockButtonAction()).toEqual(
```
Comment (tteltrab) [inline packages/fixtures/src/fields.spec.ts]: is `blockButtonAction` a good name now that it's a method on `actions`?

File: packages/jest-bolt-receiver/README.md [hunk 5]
```diff
@@ -58,34 +59,13 @@ describe('My Awesome App', () => {
 
 ### Sending Events
 
-This project uses [@slack-wrench/fixtures](../fixtures) to generate and send events. As such, this project supports all of the features supported in fixtures such as global overrides and customizing payloads. Check its documentation for details.
+We recommend you use [@slack-wrench/fixtures](../fixtures) to generate and send events. That way, you can easily generate fixtures and have as global overrides and customizing payloads. Check its documentation for details.
```
Comment (tteltrab) [inline packages/jest-bolt-receiver/README.md]:
```suggestion
We recommend you use [@slack-wrench/fixtures](../fixtures) to generate and send events. That way, you can easily generate fixtures, set global overrides, and customize payloads. Check its documentation for details.
```

**Comment by tteltrab** [comment]
this is why we can't have nice things

image

**Comment by tteltrab** [comment]
So for some reason right now github actions are being super inconsistent, failing at different steps and different jobs failing on every run.

File: packages/fixtures/src/fields.spec.ts (hunk 4 above)
Comment (barlock) [reply packages/fixtures/src/fields.spec.ts]: Probably not

File: packages/bolt-interactions/src/types/interaction.ts (hunk 2 above)
Comment (barlock) [reply packages/bolt-interactions/src/types/interaction.ts]: There isn't a compiler flag to copy hand crafted ones. We could have a post-build step that did it, but that seemed like more work than renaming.

**Pull request merged**